### ISSUE TITLE
Error loading user modules instances

### ISSUE THREAD

**Issue by RNCTX** [babelouest/glewlwyd#151]
**Describe the issue**
"Error loading user modules instances" on startup.

**To Reproduce**
Fresh install on Ubuntu 20.04

**Expected behavior**
The daemon should start up.

**Screenshots**
```
# journalctl -xe | grep glew
Jul 11 01:40:26 ngobox systemd[1]: glewlwyd.service: Scheduled restart job, restart counter is at 312.
-- Automatic restarting of the unit glewlwyd.service has been scheduled, as the result for
-- Subject: A stop job for unit glewlwyd.service has finished
-- A stop job for unit glewlwyd.service has finished.
-- Subject: A start job for unit glewlwyd.service has finished successfully
-- A start job for unit glewlwyd.service has finished successfully.
Jul 11 01:40:26 ngobox glewlwyd[111754]: Error loading user modules instances
Jul 11 01:40:26 ngobox systemd[1]: glewlwyd.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
-- An ExecStart= process belonging to unit glewlwyd.service has exited.
Jul 11 01:40:26 ngobox systemd[1]: glewlwyd.service: Failed with result 'exit-code'.
-- The unit glewlwyd.service has entered the 'failed' state with result 'exit-code'.
```

**System (please complete the following information):**
 - OS/Environment Ubuntu 20.04
 - Browser used n/a
 - Glewlwyd Version 2.3.1
 - Source installation dpkg's
 - If applicable, what option did you use to build Glewlwyd n/a

**Additional context**

```
# ls -l /usr/lib/glewlwyd
total 16
drwxr-xr-x 2 root root 4096 Jul 10 23:07 client
drwxr-xr-x 2 root root 4096 Jul 10 23:07 plugin
drwxr-xr-x 2 root root 4096 Jul 10 23:07 scheme
drwxr-xr-x 2 root root 4096 Jul 10 23:07 user
```

```
# ls -l /usr/lib/glewlwyd/user
total 188
-rw-r--r-- 1 root root 62304 Jun 26 00:23 libusermoddatabase.so
-rw-r--r-- 1 root root 36208 Jun 26 00:23 libusermodhttp.so
-rw-r--r-- 1 root root 88216 Jun 26 00:23 libusermodldap.so
```

```
# cat /etc/glewlwyd/glewlwyd.conf
...
# user_module path
user_module_path="/usr/lib/glewlwyd/user"

# client_module path
client_module_path="/usr/lib/glewlwyd/client"

# user_auth_scheme_module path
user_auth_scheme_module_path="/usr/lib/glewlwyd/scheme"

# plugin_module path
plugin_module_path="/usr/lib/glewlwyd/plugin"
...
```

```

**Comment by babelouest** [comment]
Can you post glewlwyd's log file content as well? There should be more log message to help us.

**Comment by babelouest** [comment]
It also may be related to #150, if your modules are on a filesystem that doesn't support `readdir` entirely.

This bug is fixed in the master branch. I think I'll make a new release soon just to help users with that sneaky bug...

**Comment by RNCTX** [comment]
Fixed! Mistakenly ran the initial DB migration as postgres rather than the auth db user, so it was an SQL permission issue.  Related to this, I had the log set to "file" and a log directory with permissions 0755 owned by the user/group the daemon runs as, and a log file created with 0644 permissions inside that directory owned by the same user/group, and the server did not write to it.  After changing the log to the syslog I got debug output there, though.

Have another question about reverse proxy that I'll make a new topic for.
